src/core/exec-invoke.c

```diff
@@ -2244,6 +2244,72 @@ static int build_pass_environment(const ExecContext *c, char ***ret) {
         return 0;
 }
 
+_noreturn_ static void sd_userns(int errno_pipe[2], int unshare_ready_fd,
```

Comment:
please give it a proper name. setup_private_users_child?

Reply:
Tbh I'd just drop this commit, there's no point in splitting it out anymore now that we've changed the approach.

Reply:
It helps to keep setup_private_users() smaller.
It's a static inline, what's wrong with keeping the split anyway?

Reply:
Yeah, I like the change.